### PR TITLE
Added Telemetry and bug fixes

### PR DESCRIPTION
**Pull request by ShreyasRmsft**

### CODE REVIEW

File: Tasks/VsTest/taskinputparser.ts [hunk 1]
```diff
@@ -179,15 +180,18 @@ function initTestConfigurations(testConfiguration: models.TestConfigurations) {
         }
         if (testConfiguration.vsTestVersion.toLowerCase() === 'toolsinstaller') {
             tl.debug("Trying VsTest installed by tools installer.");
-
+            ci.publishEvent( { subFeature: 'ToolsInstallerSelected', isToolsInstallerPackageLocationSet: !utils.Helper.isNullEmptyOrUndefined(tl.getVariable(constants.VsTestToolsInstaller.PathToVsTestToolVariable)) } );
```
Comment (cltshivash) [inline Tasks/VsTest/taskinputparser.ts]: don't think we need a separate CI here.. can we fold this into the main CI event ? (you do want to know other versions used as well)

File: Tasks/VsTest/taskinputparser.ts [hunk 2]
```diff
@@ -179,15 +180,18 @@ function initTestConfigurations(testConfiguration: models.TestConfigurations) {
...
+            
             testConfiguration.toolsInstallerConfig = getToolsInstallerConfiguration();
 
             // if Tools installer is not there throw.
             if(utils.Helper.isNullOrWhitespace(testConfiguration.toolsInstallerConfig.vsTestPackageLocation)) {
+                ci.publishEvent( { subFeature: 'ToolsInstallerInstallationError' } );
```
Comment (cltshivash) [inline Tasks/VsTest/taskinputparser.ts]: two CI events one after another ?

File: Tasks/VsTest/taskinputparser.ts [hunk 3]
```diff
@@ -179,15 +180,18 @@ function initTestConfigurations(testConfiguration: models.TestConfigurations) {
...
                 utils.Helper.publishEventToCi(AreaCodes.SPECIFIEDVSVERSIONNOTFOUND, 'Tools installer task did not complete successfully.', 1040, true);
                 throw new Error(tl.loc('ToolsInstallerInstallationError'));
             }
 
+            ci.publishEvent( { subFeature: 'ToolsInstallerInstallationSuccessful' } );
```
Comment (cltshivash) [inline Tasks/VsTest/taskinputparser.ts]: looks redundant to me..

File: Tasks/VsTestPlatformToolInstaller/task.json [hunk 4]
```diff
@@ -14,7 +14,11 @@
     "version": {
         "Major": 1,
         "Minor": 0,
+<<<<<<< HEAD
```
Comment (cltshivash) [inline Tasks/VsTestPlatformToolInstaller/task.json]: can you fix this ...?

File: Tasks/VsTestPlatformToolInstaller/task.loc.json [hunk 5]
```diff
@@ -14,7 +14,11 @@
   "version": {
     "Major": 1,
     "Minor": 0,
+<<<<<<< HEAD
+    "Patch": 2
+=======
```
Comment (cltshivash) [inline Tasks/VsTestPlatformToolInstaller/task.loc.json]: same..

File: Tasks/VsTestPlatformToolInstaller/vstestplatformtoolinstaller.ts [hunk 6]
```diff
@@ -5,34 +5,45 @@ import * as restm from 'typed-rest-client/RestClient';
...
+
+        ci.publishEvent('Options', { versionSelectorInput: versionSelectorInput, testPlatformVersion: testPlatformVersion } );
 
         //TODO: Add an input for cleaning up the tool cache?
         
         // Get the required version of the platform and make necessary preparation to allow its consumption down the phase 
         await getVsTestPlatformTool(testPlatformVersion, versionSelectorInput);
     }
     catch (error) {
+        ci.publishEvent('Completed', { isSetupSuccessful: 'false', error: error.message } );
```
Comment (cltshivash) [inline Tasks/VsTestPlatformToolInstaller/vstestplatformtoolinstaller.ts]: can we have additional properties indicating that it is from the task installer..?

File: Tasks/VsTestPlatformToolInstaller/task.json (hunk 4 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestPlatformToolInstaller/task.json]: hmm, had resolved this in vscode

File: Tasks/VsTestPlatformToolInstaller/vstestplatformtoolinstaller.ts (hunk 6 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestPlatformToolInstaller/vstestplatformtoolinstaller.ts]: There is a dedicated feature for tools installer

File: Tasks/VsTest/taskinputparser.ts (hunk 1 above)
Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: Other versions used we can get from our single ci event

Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: this was specifically to see if the tasks are being coupled

File: Tasks/VsTest/taskinputparser.ts (hunk 3 above)
Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: okay will remove

File: Tasks/VsTestPlatformToolInstaller/task.loc.json (hunk 5 above)
Comment (ShreyasRmsft) [reply Tasks/VsTestPlatformToolInstaller/task.loc.json]: must've happened after the latest merge, but I remember resolving and committing on vscode

File: Tasks/VsTest/taskinputparser.ts (hunk 2 above)
Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: removed one

File: Tasks/VsTest/taskinputparser.ts (hunk 1 above)
Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: folding this into the single ci events

Comment (ShreyasRmsft) [reply Tasks/VsTest/taskinputparser.ts]: actually no, leaving this here, in the dta flow the single ci will only fire when dta execution host is invoked. That may not happen due to some other error.

**Pull request merged**